tests/unit/acquisition/test_optimizer.py

```diff
@@ -116,6 +155,90 @@ def test_generate_continuous_optimizer_raises_with_invalid_init_params() -> None
         generate_continuous_optimizer(num_recovery_runs=-5)
 
 
+@pytest.mark.parametrize("num_optimization_runs", [1, 10])
+@pytest.mark.parametrize("num_recovery_runs", [1, 10])
+def test_optimize_continuous_raises_for_impossible_optimization(
+    num_optimization_runs: int, num_recovery_runs: int
+) -> None:
+    search_space = Box([-1, -1], [1, 2])
+    optimizer = generate_continuous_optimizer(
+        num_optimization_runs=num_optimization_runs, num_recovery_runs=num_recovery_runs
+    )
+    with pytest.raises(FailedOptimizationError) as e:
+        optimizer(search_space, _delta_function(10))
+    assert (
+        str(e.value)
+        == f"""
+                    Acquisition function optimization failed,
+                    even after {num_recovery_runs + num_optimization_runs} restarts.
+                    """
+    )
+
+
+@pytest.mark.parametrize("num_optimization_runs", [1, 10])
+@pytest.mark.parametrize("num_initial_samples", [1000, 5000])
+def test_optimize_continuous_uses_correctly_uses_init_params(
+    num_optimization_runs: int, num_initial_samples: int
+) -> None:
+
+    querying_initial_sample = True
+
+    def _target_fn(x: TensorType) -> TensorType:
+        nonlocal querying_initial_sample
+
+        if querying_initial_sample:  # check size of initial sample
+            assert tf.shape(x)[0] == num_initial_samples
+        else:  # check evaluations are in parallel with correct batch size
```

Comment:
is this condition ever entered?

Reply:
yep, we set`querying_initial_sample = False` in the line below